composables/zksync/useWithdrawalFinalization.ts

```diff
@@ -1,43 +1,34 @@
 import { useMemoize } from "@vueuse/core";
 import { Wallet } from "zksync-ethers";
-import IL1SharedBridge from "zksync-ethers/abi/IL1SharedBridge.json";
+import IL1Nullifier from "zksync-ethers/abi/IL1Nullifier.json";
+import { IL1AssetRouter__factory as IL1AssetRouterFactory } from "zksync-ethers/build/typechain";
 
 import { useSentryLogger } from "../useSentryLogger";
 
 import type { Hash } from "@/types";
+import type { FinalizeWithdrawalParams } from "zksync-ethers/build/types";
 
 export default (transactionInfo: ComputedRef<TransactionInfo>) => {
   const status = ref<"not-started" | "processing" | "waiting-for-signature" | "sending" | "done">("not-started");
   const error = ref<Error | undefined>();
   const transactionHash = ref<Hash | undefined>();
   const onboardStore = useOnboardStore();
   const providerStore = useZkSyncProviderStore();
+  const walletStore = useZkSyncWalletStore();
   const tokensStore = useZkSyncTokensStore();
   const { isCorrectNetworkSet } = storeToRefs(onboardStore);
   const { ethToken } = storeToRefs(tokensStore);
   const { captureException } = useSentryLogger();
 
   const retrieveBridgeAddresses = useMemoize(() => providerStore.requestProvider().getDefaultBridgeAddresses());
-
-  const retrieveChainId = useMemoize(() =>
-    providerStore
-      .requestProvider()
-      .getNetwork()
-      .then((network) => network.chainId)
-  );
+  const retrieveL1NullifierAddress = useMemoize(async () => {
+    const providerL1 = walletStore.getL1VoidSigner();
+    return IL1AssetRouterFactory.connect((await retrieveBridgeAddresses()).sharedL1, providerL1).L1_NULLIFIER();
+  });
 
   const gasLimit = ref<bigint | undefined>();
   const gasPrice = ref<bigint | undefined>();
-  const finalizeWithdrawalParams = ref<
-    | {
-        l1BatchNumber: unknown;
-        l2MessageIndex: unknown;
-        l2TxNumberInBlock: unknown;
-        message: unknown;
-        proof: unknown;
-      }
-    | undefined
-  >();
+  const finalizeWithdrawalParams = ref<FinalizeWithdrawalParams | undefined>();
```

Comment:
It seems like we don't need to save it into ref. It could be just a local variable inside of `getTransactionParams`

Reply:
Will resolve in a separate PR